src/03-a-two-state-system-in-a-quantised-field.md

```diff
@@ -0,0 +1,602 @@
+        which_mode = divmod(i,2)
+        if which_mode[1] == 0:
+            two_state = "+"
+        elif which_mode[1] == 1:
+            two_state = "-"
+        data[str(which_mode[0])+" , "+two_state] = np.zeros(len(times),dtype="complex128")
+    
+    for i, state in enumerate(states):
+        for j, psi in enumerate(state):
+            which_mode = divmod(j,2)
+            if which_mode[1] == 0:
+                two_state = "+"
+            elif which_mode[1] == 1:
+                two_state = "-"
+            data[str(which_mode[0])+" , "+two_state][i] = psi[0][0]
+
+    return pd.DataFrame(data=data, index=times)
+```
+
+The quintessential set-up for spontaneous emission is to have the field empty, i.e. no bosons $n=0$, and the two state system in it's "excited" (aka higher energy "+") state.
+
+Let's see what happens.
+
+```python
+psi0 = tensor(basis(max_bosons+1, 0), basis(2, 0))  # No bosons and two-state system is in the higher energy + state
+
+times = np.linspace(0.0, 70.0, 1000)      # simulation time
+
+result = sesolve(H, psi0, times)
+df_coupled =  states_to_df(result.states, times)
+```
+
+```python
+fig, axes = plt.subplots(nrows=1, ncols=2, figsize=(15,6))
+df_coupled.plot(title="Real part of amplitudes Re($\psi$)     (Fig 1)", ax=axes[0]);
+(df_coupled.abs()**2).plot(title="Probabilities $|\psi|^2$     (Fig 2))", ax=axes[1]);
+```
+
+It's taken us a while but we eventually arrived, as promised. We see in Fig 2 that even though we start with the field "empty" of bosons (blue line), after a while, the field is likely to have a single boson at the expense of the energy in the two state system which transitions to the lower state (red line) - the atom "spontaneously" "emits" a boson.
+
+
+You might also notice some other features:
+1. There is non-negligible chance to find the system in what appears to be a non-energy conserving state with 2 bosons and the two state system in the + state (purple line)
+2. The spontaneous emission doesn't appear to be irreversible (as we are normally taught)
+
+On point 1. To understand what's going on here, we need to take a deeper look into how the different bits of the Hamiltonian talk to each other and this will take us into the world of virtual states - it's a big enough topic to save for another tutorial.
+
+On point 2, technically spontaneous emission isn't irreversible - if you wait long enough the system will return to it's original state. However, the mode modes you have, the more places there are for the energy to go. We know from statistical physics what that means - the system will most likely be found in a high entropy state, i.e. not in our special initial condition, but in one where the energy is in the field with its uncountably many modes.
+
+We can start to get a glimpse of many mode physics by simply adding more terms to our tensor products. This gets a bit laborious so we'll make a function to do this for us:
+
+```python
+def multi_modes(number_of_modes):
```

Comment:
Probably my fault, but it would be better to write the latex equation of the Hamiltonian with multiple modes

Reply:
It's a good suggestion I'll add it now.